### PR TITLE
bug(repo sync): ignore inactive repos

### PR DESCRIPTION
**Pull request by KellyMerrick**
This closes https://github.com/go-vela/community/issues/852.

Add bool to track if webhook has been deleted from SCM:
https://github.com/go-vela/server/blob/f89e9299cf7b38ca9bf097e288769df406958ce4/scm/github/repo.go#L263-L274

And then update status to inactive in the database:
https://github.com/go-vela/server/blob/f89e9299cf7b38ca9bf097e288769df406958ce4/api/scm/sync_org.go#L140-L164

### CODE REVIEW

File: api/scm/sync.go [hunk 1]
```diff
@@ -124,7 +125,7 @@ func SyncRepo(c *gin.Context) {
 		}
 
 		// update webhook
-		err = scm.FromContext(c).Update(u, r, lastHook.GetWebhookID())
+		_, err = scm.FromContext(c).Update(u, r, lastHook.GetWebhookID())
```
Comment (ecrupper) [inline api/scm/sync.go:128]: I say we do the same think at the repo level as we do the org level (i.e. change the repo to inactive if the webhook isn't found when attempting to update)

File: scm/github/repo.go [hunk 2]
```diff
@@ -258,13 +258,20 @@ func (c *client) Update(u *library.User, r *library.Repo, hookID int64) error {
...
+		isWebhookDel = true
+	}
 
 	if err != nil {
-		return err
+		return isWebhookDel, err
 	}
 
-	return nil
+	return isWebhookDel, nil
```
Comment (ecrupper) [inline scm/github/repo.go:274]: Thoughts on shuffling this a bit to
```go
	if err != nil {
		return resp.StatusCode == http.StatusNotFound, err
	}

	return true, nil
```

That way the affirmative `true` is when the webhook does exist, and `false` is when the webhook doesn't exist. Or we could return the code itself and let the API do the logic? Not really sure the cleanest way...

**Review by ecrupper — CHANGES_REQUESTED** [review]

File: scm/github/repo.go (hunk 2 above)
Comment (KellyMerrick) [reply scm/github/repo.go]: I did implement the inverse for now, for mvp.

File: api/scm/sync.go [hunk 3]
```diff
@@ -124,13 +125,27 @@ func SyncRepo(c *gin.Context) {
 		}
 
 		// update webhook
-		err = scm.FromContext(c).Update(u, r, lastHook.GetWebhookID())
+		webhookExists, err := scm.FromContext(c).Update(u, r, lastHook.GetWebhookID())
 		if err != nil {
 			retErr := fmt.Errorf("unable to update repo webhook for %s: %w", r.GetFullName(), err)
 
 			util.HandleError(c, http.StatusInternalServerError, retErr)
```
Comment (ecrupper) [inline api/scm/sync.go:132]: I don't think we want these lines in the error block unless the error is something other than `Not Found`

File: api/scm/sync_org.go [hunk 4]
```diff
@@ -137,12 +138,28 @@ func SyncReposForOrg(c *gin.Context) {
 			}
 
 			// update webhook
-			err = scm.FromContext(c).Update(u, repo, lastHook.GetWebhookID())
+			webhookExists, err := scm.FromContext(c).Update(u, repo, lastHook.GetWebhookID())
 			if err != nil {
 				retErr := fmt.Errorf("unable to update repo webhook for %s: %w", repo.GetFullName(), err)
 
 				util.HandleError(c, http.StatusInternalServerError, retErr)
```
Comment (ecrupper) [inline api/scm/sync_org.go:145]: Same idea as above

File: api/scm/sync_org.go [hunk 5]
```diff
@@ -137,12 +138,28 @@ func SyncReposForOrg(c *gin.Context) {
...
 
+				// if webhook has been manually deleted from GitHub,
+				// set to inactive in database
+				if !webhookExists {
+
+					repo.SetActive(false)
+
+					_, err := database.FromContext(c).UpdateRepo(ctx, repo)
```
Comment (ecrupper) [inline api/scm/sync_org.go:153]: We'll want a `continue` after the successful execution of this update so we can continue updating repos if one gets tripped up by this `Not Found` business

**Review by ecrupper — CHANGES_REQUESTED** [review]

File: scm/github/repo.go [hunk 6]
```diff
@@ -258,13 +258,20 @@ func (c *client) Update(u *library.User, r *library.Repo, hookID int64) error {
...
+	}
 
 	if err != nil {
-		return err
+		return webhookExists, err
 	}
 
-	return nil
+	return webhookExists, nil
 }
```
Comment (wass3rw3rk) [inline scm/github/repo.go:275]: suggestion: we can just collapse this to `return webhookExists, err` and skip the error check.

File: api/scm/sync.go (hunk 3 above)
Comment (KellyMerrick) [reply api/scm/sync.go]: addressed

File: api/scm/sync_org.go (hunk 4 above)
Comment (KellyMerrick) [reply api/scm/sync_org.go]: addressed

File: api/scm/sync_org.go (hunk 5 above)
Comment (KellyMerrick) [reply api/scm/sync_org.go:150]: addressed

File: scm/github/repo.go (hunk 6 above)
Comment (KellyMerrick) [reply scm/github/repo.go]: addressed

**Review by ecrupper — DISMISSED** [review]

**Review by wass3rw3rk — APPROVED** [review]

**Review by ecrupper — DISMISSED** [review]

File: scm/github/repo.go [hunk 7]
```diff
@@ -258,13 +258,11 @@ func (c *client) Update(u *library.User, r *library.Repo, hookID int64) error {
...
-	_, _, err := client.Repositories.EditHook(ctx, r.GetOrg(), r.GetName(), hookID, hook)
+	_, resp, err := client.Repositories.EditHook(ctx, r.GetOrg(), r.GetName(), hookID, hook)
 
-	if err != nil {
-		return err
-	}
-
-	return nil
+	// track if webhook exists in GitHub; a missing webhook
+	// indicates the webhook has been manually deleted from GitHu
```
Comment (wass3r) [inline scm/github/repo.go:264]: RIP "b" on GitHu(b)

**Review by wass3r — DISMISSED** [review]

File: scm/github/repo.go (hunk 7 above)
Comment (ecrupper) [reply scm/github/repo.go:264]: It was manually deleted

**Review by wass3r — APPROVED** [review]

File: scm/github/repo.go (hunk 7 above)
Comment (KellyMerrick) [reply scm/github/repo.go]: bfixed it

**Review by ecrupper — APPROVED** [review]

**Pull request merged**